icepyx/core/visualization.py

```diff
@@ -0,0 +1,431 @@
+"""
+Interactive visualization of spatial extent and ICESat-2 elevations
+"""
+import backoff
+import concurrent.futures
+import dask.array as da
+import dask.dataframe as dd
+import datashader as ds
+import holoviews as hv
+from holoviews.operation.datashader import rasterize
+import intake.source.utils
+import numpy as np
+import pandas as pd
+import requests
+from tqdm import tqdm
+
+import icepyx as ipx
+
+hv.extension("bokeh")
```

Comment:
Thanks for the comments! The Binder test is great! I think `hv.extension("bokeh")` is necessary here, `matplotlib` backend is only for static map and we want interactive plot right? Yes we can put `holoviews` backend in the notebook, then we also need to do the same for `geoviews`. Not all users will have geoviews installed, so it may throw an import error if they don't have it, but this error will remind users to install geoviews themselves instead of just plot the static extent map silently in `visualize_spatial_extent()` as before, not sure about this though...any idea?

Reply:
Yes we prefer to have interactivity out of the box, but I think Bokeh is the default backend already right? I.e. users would have to explicitly set `hv.extension("matplotlib")` to get an static plot for Bokeh >= 2.0, otherwise an interactive plot should be returned by default. Maybe try removing this line and re-run your example notebook to see if you get an interactive or static map.

Reply:
If not set `hv.extension`, it will return: 

> ValueError: No plotting extension is currently loaded. Ensure you load an plotting extension with hv.extension or import it explicitly from holoviews.plotting before applying any options.

If set `hv.extension("matplotlib")`, it will have `ValueError: Tiles type not found, could not apply options.`, looks like map tiles are only supported in Bokeh, see https://holoviews.org/reference/elements/bokeh/Tiles.html. 

Also to my understanding, 

> The default plotting extension for HoloViews until a 2.0 release is Matplotlib when HoloViews will start defaulting to Bokeh (see the Plotting with Bokeh user guide).
 
2.0 release is the version requirement for Holoviews (current version is 1.14.3) rather than Bokeh, so the default is a static map from `hv.extension("matplotlib")`, Bokeh will be the default when Holoviews >= 2.0. 

```
import numpy as np
import holoviews as hv
from holoviews import opts
img = hv.Image(np.random.rand(10, 10))
fig = hv.render(img)
fig
```
This example returns a static map. 

Since we only need interactive plots, I think we could just set Bokeh as default in backend for geoviews and holoviews.

Reply:
Ah ok, thanks for the clarification! Then it's fine to keep the `hv.extension("bokeh")` line intact. I got confused with the bokeh and holoviews versions, I know bokeh is at v2 but hadn't realize holoviews is still at v1 :grimacing: Also we need the map tiles so Bokeh backend it is then!